src/components/Link/Link.scss

```diff
@@ -49,3 +49,10 @@
 .ms-Link {
   @include ms-Link;
 }
+
+// For button links we need to ensure by default that we inherit the font size. Otherwise
+// the button default font size will override.
+
+BUTTON.ms-Link {
+  font-size: inherit;
```

Comment:
I don't think this will work. Button elements have a default font-size. I tried removing it and it looks like this (notice the 2nd link is rendered as a button):

![image](https://cloud.githubusercontent.com/assets/1110944/18758100/82a36fce-80ab-11e6-9102-6b29bef44b8b.png)

Inspecting the button to identify where font-size is coming from:

![image](https://cloud.githubusercontent.com/assets/1110944/18758151/b8ecf06e-80ab-11e6-9add-bb9a9e239bf2.png)

This actually is more specific than root ms-Fabric and overriding the ms-font-m class style:

![image](https://cloud.githubusercontent.com/assets/1110944/18758170/cf2fda1c-80ab-11e6-9b6d-6f365871095b.png)